lib/mailgun_rails/deliverer.rb

```diff
@@ -21,7 +21,13 @@ def verify_ssl
     end
 
     def deliver!(rails_message)
-      response = mailgun_client.send_message build_mailgun_message_for(rails_message)
+      begin
+        response = mailgun_client.send_message build_mailgun_message_for(rails_message)
+      rescue RestClient::Exception => mailgun_error
+        json = JSON.parse(mailgun_error.http_body.to_s) rescue raise(mailgun_error)
+        mailgun_error.define_singleton_method(:message) { json["message"] }
```

Comment:
I think that bad idea to monkey patch RestClient::* classes on the fly. It is foreign classes and this changes can be counterintuitive for devs, who use API of RestClient and they can wait documented behaviour. 

For this case better to use own Exception classes.
In additional case I think about using of API from mailgun-ruby gem into this gem (mailgun-ruby gem already have implemented all required api calls and no reason why we should duplicate this logic).